File: v2/code-samples/PHP/getty_connect_api_download_p1_get_image_download_auth.php
<?php

// grab  editorial "angelina jolie" image for this example
$endpoint_search = "http://connect.gettyimages.com/v1/search/SearchForImages";

// build array to query api for images
$searchImagesArray = array (
	"RequestHeader" => array (
		"Token" => $token
	),
	"SearchForImages2RequestBody" => array (
 		"Query" => array (
			"SearchPhrase" => "angelina jolie" // search for angelina jolie
 		),
 		"Filter" => array(
 			"ImageFamilies" => array("editorial") // specify only editorial image family here
 		),
  		"ResultOptions" => array (
			"IncludeKeywords" => "false",
 			"ItemCount" => 1, // return only 25 items
 			"ItemStartNumber" => 1 // 1-based int, start at the first image
 		)
	)
);

// encode
$json = json_encode($searchImagesArray);

$httpClient = new Zend_Http_Client($endpoint_search);
$httpClient->setRawData($json, 'application/json');
$httpClient->setMethod(Zend_Http_Client::POST); // all getty api requests are POST

// Zend_Http_Response
$response = $httpClient->request();

// evaluate for success response
if ($response->isSuccessful()) {
	$body = json_decode($response->getBody()); // returns stdObject
}

// get image array from the stdObject
$images = $body->SearchForImagesResult->Images;

// for each image (in this case, 1), pull the image details from the api and get the size key to send for authorizations
foreach ($images as $image) {
	$imageId = $image->ImageId;

	// build call for this image's details, where sizeKey resides
	$endpoint_imagedetails = "http://connect.gettyimages.com/v1/search/GetImageDetails";

	// build array to query api for images
	$imageDetailsArray = array (
		"RequestHeader" => array (
			"Token" => $token
		),
		"GetImageDetailsRequestBody" => array (
			"CountryCode" => "USA", // http://en.wikipedia.org/wiki/ISO_3166-1_alpha-3#Current_codes
 			"ImageIds" => (is_array($imageId) ? $imageId : array($imageId)) // specify the image ID to get details from
 		)
	);

	// encode
	$json_imagedetails = json_encode($imageDetailsArray);

	// dispatch search, get body of response
	$imageDetailsBody = dispatchRequest($endpoint_imagedetails, $json_imagedetails);

	// get size key from the image details stdObject
	$imageDetails = $imageDetailsBody->GetImageDetailsResult->Images[0];
	$imageSizeKey = $imageDetails->SizesDownloadableImages[0]->SizeKey;

	// build request to download authorizations
	$endpoint_download_auth = "http://connect.gettyimages.com/v1/download/GetImageDownloadAuthorizations";

	// build array to query api for images
	$imageAuthorizationArray = array (
		"RequestHeader" => array (
			"Token" => $token
		),
		"GetImageDownloadAuthorizationsRequestBody" => array (
	 		"ImageSizes" => array(array (
				"ImageId" => $imageId,
				"SizeKey" => $imageSizeKey
	 		))
		)
	);

	// encode
	$json_download_auth = json_encode($imageAuthorizationArray);

	$httpClient = new Zend_Http_Client($endpoint_download_auth);
	$httpClient->setRawData($json_download_auth, 'application/json');
	$httpClient->setMethod(Zend_Http_Client::POST); // all getty api requests are POST

	// Zend_Http_Response
	$response = $httpClient->request();

	// evaluate for success response, get the body of data returned for download authorization
	if ($response->isSuccessful()) {
		$imageDownloadAuthBody = json_decode($response->getBody()); // returns stdObject
	}

	print_r($imageDownloadAuthBody);
}
?>

File: v2/code-samples/Perl/getty_connect_api_download_p1_get_image_download_auth.pl
#!/usr/bin/perl -w
# $Rev: 1 $
# $Application: Getty Connect $
# $Date: 2012-02-21 $
#
# Requires Json
#
use strict;
use warnings;
use JSON;
use CGI;
use LWP::UserAgent;

#set params
my $token   = "token";
my $imageId = "";
my $sizeKey = "";
my $url = "http://connect.gettyimages.com/v1/download/GetImageDownloadAuthorizations";

# create array of data for request
my $json_data = {
	"RequestHeader" => {
		"CoordinationId" => "",
		"Token"          => $token
	},
	"GetImageDownloadAuthorizationsRequestBody" => {
		"ImageIds" => {
			"ImageId" => $imageId,
			"SizeKey" => $sizeKey
		}
	}
};

#encode data to json
$json_data = encode_json $json_data;

# set method to POST
my $req = HTTP::Request->new( POST => $url );
$req->content_type('application/json');
$req->content($json_data);

# dispatch request, returns
my $ua       = LWP::UserAgent->new;
my $response = $ua->request($req);

#decode data to json
$response = decode_json $response->content;

#print response
print( $response. "\n" );

